parser/statement_parsing.go

```diff
@@ -63,11 +64,15 @@ func (p *Parser) parseDefMethodStatement() *ast.DefStatement {
 	}
 	// def foo
 	if p.peekTokenAtSameLine() { // def foo(), next token is ( and at same line
-		if !p.expectPeek(token.LParen) {
-			return nil
+		if p.peekTokenIs(token.LParen) {
+			if !p.expectPeek(token.LParen) {
```

Comment:
If you confirmed next token is `LParen` above, why check it again here? This will always be false.